### PR TITLE
feat: allow caching of v4 and PII tokens

### PR DESCRIPTION
**Pull request by Alex-Joseph**
https://rewardops.atlassian.net/browse/MX-1110

- add validation to `auth.getToken` to ensure `tokenId` param is valid
- create separate tokens for v4 endpoints (default) and PII endpoints
- the caller is responsible to ensure the `tokenId` is correct

**TODO**
- refactor the `api` module to be used in the `order-recipient` module.

### CODE REVIEW

File: lib/auth.js [hunk 1]
```diff
@@ -12,11 +12,14 @@ const urls = require('./urls');
 const { generateBasicAuthToken } = require('./utils/auth');
 
 /**
- * Token lock status
+ * Token lock status for v4 and PII
  *
  * @private
  */
-let tokenLocked = false;
+const tokenLockStatus = {
```
Comment (Alex-Joseph) [inline lib/auth.js:19]: Covert to `new Map([['v4', false], ['pii', false])`?

Comment (shamrt) [reply lib/auth.js:19]: Why a `Map`?

File: test/auth.test.js [hunk 2]
```diff
@@ -534,19 +534,58 @@ describe('RO.auth', () => {
...
 
     it('should listen to the invalidateToken event', () => {
-      RO.auth.token = { imStillA: 'token' };
+      RO.auth.tokens = { v4: { token: 'v4Token' }, pii: { token: 'piiToken' } };
+
+      emitter.emit('invalidateToken', 'v4');
+
+      expect(RO.auth.tokens.v4).toEqual({});
+      expect(RO.auth.tokens.pii).toEqual({ token: 'piiToken' });
+    });
```
Comment (shamrt) [inline test/auth.test.js:551]: Same here

File: test/test-helpers/mock-config.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ const mockConfig = ({
   timeout = 20000,
   verbose = true,
   supportedLocales = undefined,
-}) => ({
+} = {}) => ({
```
Comment (shamrt) [inline test/test-helpers/mock-config.js:24]: 👍

**Review by shamrt — COMMENTED** [review]
Some questions

**Comment by Alex-Joseph** [comment]
https://rewardops.atlassian.net/browse/MX-1122 will solve this edge case
